### PR TITLE
Fixes for Swagger generation (required fields and security)

### PR DESCRIPTION
**Pull request by bradfriedman**
This fixes the issue brought up in https://github.com/cloudendpoints/endpoints-python/issues/21, as well as fixes an issue with Swagger config for API keys.

### CODE REVIEW

File: endpoints/swagger_generator.py [hunk 1]
```diff
@@ -547,12 +547,21 @@ def __definitions_descriptor(self):
...
+          key_result['type'] = def_value['type']
+        if 'properties' in def_item_keys:
+          for prop_key, prop_value in def_value['properties'].items():
+            if isinstance(prop_value, dict) and 'required' in prop_value:
+              required_keys.add(prop_key)
+              del prop_value['required']
+          key_result['properties'] = def_value['properties']
+        # Add in the required fields, if any
+        if required_keys:
+          key_result['required'] = sorted(list(required_keys))
```
Comment (tangiel) [inline endpoints/swagger_generator.py]: I believe this is redundant, you should just need sorted(required_keys).

File: endpoints/swagger_generator.py [hunk 2]
```diff
@@ -547,12 +547,21 @@ def __definitions_descriptor(self):
     # Filter out any keys that aren't 'properties' or 'type'
     result = {}
     for def_key, def_value in self.__parser.schemas().iteritems():
-      prop_keys = def_value.keys()
-      if 'properties' in prop_keys or 'type' in prop_keys:
+      def_item_keys = def_value.keys()
+      if 'properties' in def_item_keys or 'type' in def_item_keys:
```
Comment (tangiel) [inline endpoints/swagger_generator.py]: Can't you just do:

if 'properties' in def_item or 'type' in def_item? You do the same thing below with 'required' in prop_value. Ditto below.

File: endpoints/swagger_generator.py [hunk 3]
```diff
@@ -547,12 +547,21 @@ def __definitions_descriptor(self):
...
         key_result = {}
-        for key in ('properties', 'type'):
-          if key in prop_keys:
-            key_result[key] = def_value[key]
+        required_keys = set()
+        if 'type' in def_item_keys:
+          key_result['type'] = def_value['type']
+        if 'properties' in def_item_keys:
+          for prop_key, prop_value in def_value['properties'].items():
+            if isinstance(prop_value, dict) and 'required' in prop_value:
```
Comment (tangiel) [inline endpoints/swagger_generator.py]: Is there a case where prop_value isn't going to be a dict?

Comment (bradfriedman) [inline endpoints/swagger_generator.py]: Probably not -- just threw that clause in as a failsafe.

File: endpoints/swagger_generator.py (hunk 2 above)
Comment (bradfriedman) [inline endpoints/swagger_generator.py]: Yep, I can change it to 'properties' in def_value, etc. Not sure about 'required' in prop_value, though. That's a dict in and of itself.

File: endpoints/swagger_generator.py (hunk 1 above)
Comment (bradfriedman) [inline endpoints/swagger_generator.py]: True, changed.

**Pull request merged**